### PR TITLE
Milestone 3

### PR DESCRIPTION
**Pull request by tiagoit**
- Secure all routes but auth ones.
- Home page with Users/ Posts listing
- Create post route
- User profile route

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+class PostsController < ApplicationController
+  def index
```
Comment (mosaaleb) [inline app/controllers/posts_controller.rb]: - [ ] Index displays posts in a timeline (recent to older) -- We can achieve this by using [default scope](https://apidock.com/rails/ActiveRecord/Base/default_scope/class) in `post.rb`. An example could be [this](https://stackoverflow.com/questions/3393687/default-sort-order-for-a-rails-model).

- [ ] **[OPTIONAL]** We could just say `@posts = current_user.posts`

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    @posts = Post.where(user_id: current_user.id)
```
Comment (mosaaleb) [inline app/controllers/posts_controller.rb:3]: - [x] Index displays posts in a timeline (recent to older) -- We can achieve this by using [default scope](https://apidock.com/rails/ActiveRecord/Base/default_scope/class) in `post.rb`. An example could be [this](https://stackoverflow.com/questions/3393687/default-sort-order-for-a-rails-model).

- [ ] **[OPTIONAL]** We could just say `@posts = current_user.posts`

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    else
+      flash[:danger] = 'Something went wrong, please try again.'
+      render :new
+    end
+  end
+
+  def show
+  end
+
+  def edit
```
Comment (mosaaleb) [inline app/controllers/posts_controller.rb]: - [ ] **[OPTINOAL]** Let's continue implementing edit and delete feature for posts.

File: app/views/posts/_post.html.erb [hunk 4]
```diff
@@ -0,0 +1 @@
+<%= post.content %>
```
Comment (mosaaleb) [inline app/views/posts/_post.html.erb]: - [x] Post is always displayed with content and author -- Let's add the author

**Review by mosaaleb — CHANGES_REQUESTED** [review]
## Well done @ferreirati :clap:

- [x] This is a collaborative project, but I can only only you in the commit history, have you reached the student success about this?

- [x] UsersController#index displays all users -- Let's add users controller to list all users.


- [x] We still haven't added any tests yet. Please lets add rspec **model** tests, and capybara **integration** tests for post creation.


#### Please handle these requirements and submit a new code review.

##### Muhammad Ebeid

**Review by Ispirett — APPROVED** [review]
# status Approved :+1:  :1st_place_medal: 
Great work!! done here folks @ferreirati  @oscardelalanza

**Pull request merged**